src/chi-vue/src/components/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,156 @@
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import { uuid4 } from '../../utils/utils';
+import { ACTIVE_CLASS, LIGHT_CLASS, TOOLTIP_CLASSES, TRANSITIONING_CLASS } from '../../constants/classes';
+import { TOOLTIP_EVENTS } from '../../constants/events';
+import { TooltipPositions } from '../../constants/types';
+import { ThreeStepsAnimation } from '../../utils/ThreeStepsAnimation';
+import { TOOLTIP_ANIMATION_DELAY as ANIMATION_DELAY, ANIMATION_DURATION } from '../../constants/constants';
+import { createPopper } from '@popperjs/core';
+
+@Component
+export default class Tooltip extends Vue {
+  @Prop() message!: string;
+  @Prop() color!: string;
+  @Prop({ default: 'top' }) position!: TooltipPositions;
+
+  _tooltipElement!: JSX.Element | null;
+  _tooltipElementNode!: HTMLElement;
+  _shown!: boolean;
+  tooltipElementClasses!: string[];
+  _uuid!: string;
+  _animationTimeout!: number;
+  _animation!: ThreeStepsAnimation;
+
+  constructor() {
+    super();
+    this.tooltipElementClasses = [
+      TOOLTIP_CLASSES.TOOLTIP_ELEMENT,
+      this.color === 'light' ? LIGHT_CLASS : ''
+    ];
+  }
+
+  beforeMount() {
+    this._uuid = uuid4();
+
+    this._tooltipElement = (
+      <div id={this._uuid} class={this.tooltipElementClasses.join(' ')}>
+        <span>{this.message}</span>
+      </div>
+    );
+  }
+
+  show() {
+    this.$emit(TOOLTIP_EVENTS.SHOW);
+    if (this._animation && !this._animation.isStopped()) {
+      this._animation.stop();
+    }
+
+    if (this._tooltipElementNode) {
+      if (!this._shown) {
```

Comment:
One less 'if' :)
```suggestion
    if (this._tooltipElementNode && !this._shown) {
```

Reply:
Makes sense, thank you 😉

Reply:
I changed only those lines ignoring the indentation of other lines to makes it easier to review the commit. Once the functionality is approved, I will make lint it to fix indentation and other linting issues.